pkg/node/connection_pool.go

```diff
@@ -88,12 +90,23 @@ func (pool *ConnectionPool) Dial(ctx context.Context, n *pb.Node) (pb.NodesClien
 	}
 	pool.mu.Unlock()
 
+	ticker := time.NewTicker(500 * time.Millisecond)
+	defer ticker.Stop()
 	conn.dial.Do(func() {
 		conn.grpc, conn.err = pool.tc.DialNode(ctx, n)
```

Comment:
Instead add argument `grpc.WithBlocking()` to wait for the connection to be established.

Reply:
done